pkg/registry/networking/ingress/strategy.go

```diff
@@ -93,7 +97,15 @@ func (ingressStrategy) Validate(ctx context.Context, obj runtime.Object) field.E
 }
 
 // WarningsOnCreate returns warnings for the creation of the given object.
-func (ingressStrategy) WarningsOnCreate(ctx context.Context, obj runtime.Object) []string { return nil }
+func (ingressStrategy) WarningsOnCreate(ctx context.Context, obj runtime.Object) []string {
+	var warnings []string
+	ingress := obj.(*networking.Ingress)
+	_, annotationIsSet := ingress.Annotations[annotationIngressClass]
+	if annotationIsSet && ingress.Spec.IngressClassName != nil {
+		warnings = append(warnings, "ingressClass annotation and IngressClassName should not be set at the same time")
```

Comment:
What is the thing we want to warn about, really?  Having them both set, but different, will fail.

So they have them both set to the same value - why should we warn about that?

If anything, we should warn about the case where they have annotation and NOT class name.  Or simply about having the annotation at all ("deprecated"), but that feels pointless if the class name is set.

So I'd change this to

```
if annotationIsSet && ingress.Spec.IngressClassName == nil {
		warnings = append(warnings, "annotation %q is deprecated, please use 'spec.ingressClassName' instead")
```

With the %q being the full annotation name